### ISSUE TITLE
Make it easier to navigate built-in documentation while following references to methods/properties

### ISSUE THREAD

**Issue by Xrayez** [godotengine/godot-proposals#2248]
### Describe the project you are working on
[Goost](https://github.com/goostengine/goost) Godot Engine Extension.

### Describe the problem or limitation you are having in your project

Having ability to jump between different sections in the documentation is a great feature, and works mostly well if you jump between classes. But if you have a relatively small class with lots of inner references, or the methods reside at the bottom of the list, I find it difficult to quickly identify the method/property of interest that way, because those references don't end up being placed at the top of the documentation window, so this breaks expectations, therefore makes the workflow slower.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I think that highlighting relevant sections would be enough to make it easier to navigate cross-references and inner references with long jumps.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![image](https://user-images.githubusercontent.com/17108460/107052261-1b15ae00-67d6-11eb-91a2-35e0a981e06b.png)

Notice highlighted method description when clicking on the method reference above:

![godot_docs_highlight](https://user-images.githubusercontent.com/17108460/107052571-819acc00-67d6-11eb-8611-d4e19f2ee85a.png)

Fading away the highlighted area is also acceptable.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
It's a core documentation feature.

### Is there a reason why this should be core and not an add-on in the asset library?
It's a core documentation feature.

**Comment by Calinou** [comment]
Good idea, but I don't know how we could implement this given the current feature set in RichTextLabel.

**Comment by KoBeWi** [comment]
We could just display a ColorRect over the relevant area.

**Comment by Mickeon** [comment]
Note that without https://github.com/godotengine/godot-proposals/issues/7616 the implementation may be more "hacky" than desirable, but still doable.
